SyslogNet.Client/Transport/SyslogTcpSender.cs

```diff
@@ -118,4 +129,14 @@ public void Dispose()
 			}
 		}
 	}
+
+	static class MemoryStreamExtensions
+	{
+		private static readonly byte[] WriteByteAsyncBuffer = new byte[1];
+		public static Task WriteByteAsync(this MemoryStream stream, byte value)
+		{
+			WriteByteAsyncBuffer[0] = value;

```

Comment:
Agree, but this class doesn't seem to be meant to be thread-safe (the other methods of the class are not thread-safe either).